docs/index.rst

```diff
@@ -27,3 +25,17 @@ Contents:
    api/index
    GitHub <https://github.com/twisted/twisted>
    PyPI <https://pypi.org/project/twisted>
+
+
+Sponsors
```

Comment:
I have also added the sponsors on the documentation start page.

I guess that it's ok... it should not be too much spam

It's at the bottom, so the focus on the documentation site is still to provide documentation for end users


but I think that we should provide enough benefits for companies to consider sponsoring Twisted